base/sim/FairGenericStack.h

```diff
@@ -98,8 +98,11 @@ class FairGenericStack : public TVirtualMCStack
     virtual void Register() {}
 
     virtual TClonesArray* GetListOfParticles() { return 0; }
-    virtual void SetParticleArray(TClonesArray* partArray) {}
-    virtual void SetParticleArray(TClonesArray* partArray, Int_t partFrom, Int_t partTo) {}
+    virtual void SetParticleArray(__attribute__((unused)) TClonesArray* partArray) {}
+    virtual void SetParticleArray(__attribute__((unused)) TClonesArray* partArray,
+                                  __attribute__((unused)) Int_t partFrom,
+                                  __attribute__((unused)) Int_t partTo)
+    {}
```

Comment:
use `[[gnu::unused]]` or `[[maybe_unused]]` instead of non-standard attribute syntax

Reply:
or use

 virtual void SetParticleArray(TClonesArray*) {}
virtual void SetParticleArray(TClonesArray*, Int_t, Int_t) {}

Reply:
Thanks! Ok I will use **[[maybe_unused]]**

Reply:
Note that `[[maybe_unused]]` is C++17

Reply:
> or use
> 
> virtual void SetParticleArray(TClonesArray*) {}
> virtual void SetParticleArray(TClonesArray*, Int_t, Int_t) {}

In some cases it did not work!

Reply:
I change it to  gnu-unused as we are still requiring C++11 and not 17.